cmake/DD4hepBuild.cmake

```diff
@@ -71,11 +71,22 @@ macro(dd4hep_set_compiler_flags)
     add_definitions(-DDD4HEP_USE_STDCXX=11)
   endif()
 
+  set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -pedantic -Wshadow -Wformat-security -Wno-long-long -Wdeprecated")
 
-  IF(("${CMAKE_CXX_COMPILER_ID}" EQUAL "GNU") OR ("${CMAKE_CXX_COMPILER_ID}" EQUAL "Clang"))
+  CHECK_CXX_COMPILER_FLAG("-fdiagnostics-color=always" FLAG_COLOR)
+  if (FLAG_COLOR)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=always")
+  endif()
+   
+  IF( "${CMAKE_CXX_COMPILER_ID}" EQUAL "Clang" )
     SET(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined")
   ENDIF()
 
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set( CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
+    set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--no-undefined -pthread")
+  endif()
+
```

Comment:
Regarding pthread, I would suggest using the [CMake function](https://cmake.org/cmake/help/v3.0/module/FindThreads.html) `find_package (Threads)` and the do `target_link_libraries (DDApp ${CMAKE_THREAD_LIBS_INIT})`. Along the lines discuses [here](http://stackoverflow.com/questions/1620918/cmake-and-libpthread) and [here](http://stackoverflow.com/questions/5395309/cmake-and-threads)